app/controllers/lectures_controller.rb

```diff
@@ -23,6 +23,15 @@ def edit
                               Time.parse(ENV['RAILS_CACHE_ID'])].max)
       eager_load_stuff
     end
+
+    # emergency link -> prefill form
+    status = @lecture.emergency_link_status
+    link = @lecture.emergency_link
+    if status == "lecture_link"
```

Comment:
You mean, use
`status = @lecture.emergency_link_status_for_database`
and
`if status == Lecture.emergency_link_statuses[:lecture_link]`?
Or is there a better way?

Reply:
I'm not sure what you mean by `emergency_link_status_for_database`. I meant the statement
```rb
if status == "lecture_link"
```

should be replaced by what you provided in your comment:
```rb
if status == Lecture.emergency_link_statuses[:lecture_link]
```

This way, we will encounter an error if the string for `lecture_link` is changed later on and this code occurrency is overlooked during the refactor. Otherwise (with just the string), it might not be noticed and the check just fails since the `status` is not that specific string anymore.

Reply:
The thing is, `Lecture.emergency_link_statuses[:lecture_link]` returns the integer associated to `lecture_link` while `@lecture.emergency_link_status` returns the string representation. So it doesn't make sense to compare them. Using `emergency_link_status_for_database` also returns an integer such that we can now compare both values.

Reply:
done in 57ba6e69b6